packages/app/pages/api/posts/[postId]/upvote.ts

```diff
@@ -1,9 +1,9 @@
+import { NextApiRequest, NextApiResponse } from 'next';
 import supabase from "@/lib/supabaseClient";
 
-export default async function handler(req: any, res: any) {
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
   if (req.method === "POST") {
-    // TODO: verify upvoter address using signed message
-    const { error } = await supabase.rpc("upvote", {
+    const { error } = await supabase.rpc("upordownvote", {
```

Comment:
we should check the user is authenticated (`req.session.siwe`) before allowing the vote to happen, and also switch from using the `req.body.upvoter` to `req.session.siwe.address`

Reply:
Fixed these!